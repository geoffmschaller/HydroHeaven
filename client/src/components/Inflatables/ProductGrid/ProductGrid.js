import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './ProductGrid.module.css';
import { getBrandFromId, getNameFromId } from '../../../utils/StringHelpers';

class ProductGrid extends React.Component {
	render() {
		let prods = [];
		if (this.props.count > 0) {
			for (let i = 0; i < this.props.count; i += 1) {
				prods.push(this.props.productData[Math.floor(Math.random() * this.props.productData.length)]);
			}
		} else {
			prods = this.props.productData;
		}

		return (
			<div className={styles.featured}>
				{
					!this.props.click
						? prods.map((prod, index) => (
							<div className={styles.bbq} key={index}>
								<img src={prod.image} alt="" />
								<div className={styles.title}>{getNameFromId(prod.id)}</div>
								<div className={styles.brand}>
									By
									{' '}
									{getBrandFromId(prod.id)}
								</div>
								<Link to={this.props.link + prod.id}>
									<button type="button">View Details</button>
								</Link>
								<div className="clear" />
							</div>
						))
						: prods.map((prod, index) => (
							<div className={styles.bbq} key={index}>
								<img src={prod.image} alt="" />
								<div className={styles.title}>{getNameFromId(prod.id)}</div>
								<div className={styles.brand}>
									By
									{' '}
									{getBrandFromId(prod.id)}
								</div>
								<Link to={this.props.link + prod.id} onClick={() => this.props.click(prod.id)}>
									<button type="button">View Details</button>
								</Link>
								<div className="clear" />
							</div>
						))
				}
			</div>
		);
	}
}

ProductGrid.propTypes = {
	productData: PropTypes.arrayOf(PropTypes.shape({
		id: PropTypes.string.isRequired,
		image: PropTypes.string.isRequired,
	})).isRequired,
	link: PropTypes.string.isRequired,
	count: PropTypes.number.isRequired,
	click: PropTypes.func.isRequired,
};

export default ProductGrid;
