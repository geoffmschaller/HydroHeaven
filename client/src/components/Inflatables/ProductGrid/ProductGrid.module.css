.featured {
  width: 100%;
  display: block; }
  .featured .bbq {
    width: 25%;
    padding: 0 20px;
    float: left;
    margin-bottom: 30px; }
    .featured .bbq img {
      width: 100%;
      margin-bottom: 10px; }
    .featured .bbq .title {
      font-size: 18px;
      color: #566573; }
    .featured .bbq .brand {
      font-size: 11px;
      margin-bottom: 20px;
      color: #abb2b9; }
    .featured .bbq button {
      color: #85c1e9;
      border: 2px solid #85c1e9;
      float: right; }
      .featured .bbq button:hover {
        color: white;
        background-color: #85c1e9; }

@media (max-width: 800px) {
  .featured {
    display: block; }
    .featured .bbq {
      width: 50%;
      float: left;
      margin-bottom: 30px; } }

@media (min-width: 801px) and (max-width: 1050px) {
  .featured {
    display: block; }
    .featured .bbq {
      width: 33%;
      float: left;
      margin-bottom: 30px; } }
