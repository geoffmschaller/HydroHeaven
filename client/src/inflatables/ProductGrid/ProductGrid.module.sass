@import "src/static/sass/config"

$name-color: #566573
$brand-color: #abb2b9
$button-color: #85c1e9

.featured
	width: 100%
	display: block

	.bbq
		width: 25%
		padding: 0 20px
		float: left
		margin-bottom: 30px

		img
			width: 100%
			margin-bottom: 10px

		.title
			font-size: 18px
			color: $name-color

		.brand
			font-size: 11px
			margin-bottom: 20px
			color: $brand-color

		button
			width: 100%
			background-color: transparent
			color: $button-color
			border: 2px solid $button-color
			float: right

			&:hover
				color: white
				background-color: $button-color

@media(max-width: 800px)
	.featured
		display: block

		.bbq
			width: 50%
			float: left
			margin-bottom: 30px

@media (min-width: 801px) and (max-width: 1050px)
	.featured
		display: block

		.bbq
			width: 33%
			float: left
			margin-bottom: 30px