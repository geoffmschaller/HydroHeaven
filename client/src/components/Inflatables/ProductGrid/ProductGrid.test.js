import React from 'react';
import ProductGrid from "./ProductGrid";
import SPA_IMAGE from '../../../static/images/spas/250.png';

const product = [
	{
		id: '1234',
		image: SPA_IMAGE
	}
];

const wrapper = <ProductGrid productData={product} link={"/fake-link"} count={4}/>;

describe("Product Grid Inflatable", () => {
	it("Renders Correctly", () => {
		expect(wrapper).toMatchSnapshot();
	});
});
