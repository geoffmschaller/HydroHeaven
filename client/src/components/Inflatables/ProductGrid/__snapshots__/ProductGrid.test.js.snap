// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Grid Inflatable 1`] = `
<ProductGrid
  count={4}
  link="/fake-link"
  productData={
    Array [
      Object {
        "id": "1234",
        "image": "250.png",
      },
    ]
  }
/>
`;
